### PR TITLE
Modify `ListValidatorAssignments` to use new state service

### PR DESCRIPTION
**Pull request by terencechain**
Note: Let's wait until v0.11 gets into master then merge this.

part of #5350 

This PR deprecated the archival usages for RPC end point `ListValidatorAssignments` and hooked the state logic to using the new state service. The result is a much cleaner design with the deletions of `archivedValidatorCommittee` and `archivedCommitteeData`

Tested runtime:
Screen Shot 2020-04-09 at 12 57 51 PM

### CODE REVIEW

File: beacon-chain/rpc/beacon/assignments.go [hunk 1]
```diff
@@ -49,11 +39,12 @@ func (bs *Server) ListValidatorAssignments(
 		requestedEpoch = q.Epoch
 	}
 
-	if requestedEpoch > helpers.CurrentEpoch(headState) {
```
Comment (0xKiwi) [inline beacon-chain/rpc/beacon/assignments.go:52]: Using time to detect what epoch is current instead of the head?

Comment (terencechain) [reply beacon-chain/rpc/beacon/assignments.go:52]: Correct, that's what I updated to. See below 👇 
```go
	currentEpoch := helpers.SlotToEpoch(bs.GenesisTimeFetcher.CurrentSlot())
```

File: beacon-chain/rpc/beacon/assignments.go [hunk 2]
```diff
@@ -49,18 +40,24 @@ func (bs *Server) ListValidatorAssignments(
...
 		return nil, status.Errorf(
 			codes.InvalidArgument,
 			"Cannot retrieve information about an epoch in the future, current epoch %d, requesting %d",
-			helpers.CurrentEpoch(headState),
+			currentEpoch,
 			requestedEpoch,
 		)
 	}
 
+	requestedState, err := bs.StateGen.StateBySlot(ctx, helpers.StartSlot(requestedEpoch))
```
Comment (nisdas) [inline beacon-chain/rpc/beacon/assignments.go:53]: why not just use this over here ?
```
bs.GenesisTimeFetcher.CurrentSlot()
```

Comment (terencechain) [reply beacon-chain/rpc/beacon/assignments.go:53]: User has option to choose to request an arbitrary slot that's less than current slot. See L39

```go
case *ethpb.ListValidatorAssignmentsRequest_Epoch:
		requestedEpoch = q.Epoch
	}
```

**Pull request merged**